File: app/src/main/java/com/practice/forecast/ui/map/MapContract.java
package com.practice.forecast.ui.map;

import com.google.android.gms.maps.model.LatLng;
import com.practice.forecast.ui.arch.Contract;
import com.practice.weathermodel.pojo.City;

import java.util.List;

import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.LiveData;
import io.reactivex.subjects.BehaviorSubject;

public interface MapContract {
    interface BaseMapViewModel extends LifecycleObserver {
        LiveData<List<City>> getCitiesWeather();

        LiveData<LatLng> getCurrentLocation();

        LiveData<Boolean> getConnectivityState();

        void saveLocation(LatLng location);

        void showCities(BehaviorSubject<String> coordinatesObservable);

        void findCurrentLocation(boolean hasPermission);
    }

    interface Host extends Contract.Host {
        void openDetailFragment(String cityName, int cityId);

        void closeApp();
    }
}

File: app/src/main/java/com/practice/forecast/ui/map/MapContract.kt
package com.practice.forecast.ui.map

import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.LiveData
import com.google.android.gms.maps.model.LatLng
import com.practice.forecast.ui.arch.Contract
import com.practice.weathermodel.pojo.City
import io.reactivex.subjects.BehaviorSubject

interface MapContract {
    interface BaseMapViewModel : LifecycleObserver {
        fun getCitiesWeather(): LiveData<List<City>>
        fun getCurrentLocation(): LiveData<LatLng?>
        fun getConnectivityState(): LiveData<Boolean?>
        fun saveLocation(location: LatLng)
        fun showCities(coordinatesObservable: BehaviorSubject<String>)
        fun findCurrentLocation(hasPermission: Boolean)
    }

    interface Host : Contract.Host {
        fun openDetailFragment(cityName: String?, cityId: Int)
        fun closeApp()
    }
}